kernel/kernel-api/src/main/java/io/delta/kernel/exceptions/ChecksumAlreadyExistsException.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.exceptions;
+
+import static java.lang.String.format;
+
+import io.delta.kernel.annotation.Evolving;
+
+/**
+ * Thrown when trying to create a checksum at version {@code v}, but there already exists a checksum
+ * at version {@code v}.
+ *
+ * @since 3.4.0
+ */
+@Evolving
+public class ChecksumAlreadyExistsException extends KernelException {
```

Comment:
Q: Do we need this instead of FileAlreadyExistsException?

Reply:
I removed this exception, we will not need to throw this per discussions in
https://github.com/delta-io/delta/pull/4440#discussion_r2056599017